File: pset6/credit/credit.py
from cs50 import get_int


number = get_int("Number: ")

odd_total = 0
even_total_double = 0

n = number
count = 0
upper_two_digits = 0

while n > 0:
    digits = n % 10
    if count % 2 == 0:
        odd_total += digits
    else:
        j = 2 * digits
        if j >= 10:
            k = j % 10
            j = int((j - k) / 10)
            even_total_double += (j + k)
        else:
            even_total_double += j

    n = int((n - digits) / 10)
    count += 1

    if n < 100 and n > 9:
        upper_two_digits = n


total = even_total_double + odd_total

if total % 10 == 0:
    upper_number = 0
    if upper_two_digits == 34 or upper_two_digits == 37:
        if count == 15:
            print("AMEX")
        else:
            print("INVALID")
        upper_number += 1
    if upper_two_digits - (upper_two_digits % 10) == 40:
        if count == 13 or count == 16:
            print("VISA")
        else:
            print("INVALID")
        upper_number += 1
    if upper_two_digits == 51 or upper_two_digits == 52 or upper_two_digits == 53 or upper_two_digits == 54 or upper_two_digits == 55:
        if count == 16:
            print("MASTERCARD")
        else:
            print("INVALID")
        upper_number += 1
    if upper_number == 0:
        print("INVALID")
else:
    print("INVALID")

File: pset1/credit/credit.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    //receive number
    long number = get_long("Number: ");

    int odd_total = 0;
    int even_total_double = 0;

    //Luhn's Algorithm
    long n = number;
    int count = 0;
    int upper_two_digits;
    while (n > 0)
    {
        int digits = n % 10;
        if (count % 2 == 0)
        {
            odd_total += digits;
        }
        else
        {
            int j = 2 * digits;
            if (j >= 10)
            {
                int k = j % 10;
                j = (j - k) / 10;
                even_total_double += (j + k);
            }
            else
            {
                even_total_double += j;
            }
        }
        n = (n - digits) / 10;
        count++;
        //receive upper two digits
        if (n < 100 && n > 9)
        {
            upper_two_digits = n;
        }
    }

    //research what your credit number card is and whether it is valid or not
    int all = even_total_double + odd_total;

    if (all % 10 == 0)
    {
        switch (upper_two_digits)
        {
            case 34:
            case 37:
                //AMEX number digits are 15
                if (count == 15)
                {
                    printf("AMEX\n");
                }
                else
                {
                    printf("INVALID\n");
                }
                break;
            case 40:
            case 41:
            case 42:
            case 43:
            case 44:
            case 45:
            case 46:
            case 47:
            case 48:
            case 49:
                //VISA number digits are 13 or 16
                if (count == 13 || count == 16)
                {
                    printf("VISA\n");
                }
                else
                {
                    printf("INVALID\n");
                }
                break;
            case 51:
            case 52:
            case 53:
            case 54:
            case 55:
                //MASTERCARD number digits are 16
                if (count == 16)
                {
                    printf("MASTERCARD\n");
                }
                else
                {
                    printf("INVALID\n");
                }
                break;
            default:
                printf("INVALID\n");
                break;
        }
    }
    else
    {
        printf("INVALID\n");
    }
}